content/helpers.ts

```diff
@@ -384,6 +393,58 @@ export function getCombinationBetweenTwoItems(sourceItem: ItemConfig, targetItem
   return result;
 }
 
+// recipe functions
+export function getAllRecipesForInteraction(interaction: Interaction): Recipe[] {
+  const recipes: Recipe[] = [];
+
+  for (const recipe of allRecipes) {
+    if (recipe.interaction === interaction)
+      recipes.push(recipe as Recipe);
+  }
+
+  recipes.sort((a, b): number => {
+    let aDescriptorTotal = 0;
+    a.ingredients.forEach(ingredient => aDescriptorTotal += ingredient.level);
+
+    let bDescriptorTotal = 0;
+    b.ingredients.forEach(ingredient => bDescriptorTotal += ingredient.level);
+
+    return bDescriptorTotal - aDescriptorTotal;
+  });
```

Comment:
this should probably be [`lodash.sortBy`](https://lodash.com/docs/4.17.15#sortBy) and a combination with `sumBy` - `sortBy(r => sumBy(r.ingredients.map(i => i.level)))`